### PR TITLE
Updating hub deployment information

### PR DESCRIPTION
**Pull request by choldgraf**
Updates the hub deployment information as well as the template we use to accomplish this, with the following goals:

1. De-duplicate some information
2. Provide more explanation for what fields mean, and why they're needed
3. Create more pointers between the issue template and docs
4. Re-organize the "add a new hub" docs so they're in nested sections, since they fall under the same conceptual topic.
5. Renames "Community Champion" to "Community Representative" since it is a more descriptive term.

closes https://github.com/2i2c-org/pilot-hubs/issues/527

### CODE REVIEW

**Comment by choldgraf** [comment]
I suspect this needs another round or two of edits before merging, but putting this PR up there in case others have thoughts on how it could be better. Maybe in particular @sgibson91 since she recently went through the hub deploy process.

I'd love comments on:

- Whether this is the right place for this kind of information
- Whether it will be intuitive / useful to have this combination of "issue template + docs"
- Whether we're missing any important information

**Review by sgibson91 — COMMENTED** [review]
Great start! I left a few comments. I know it will involve some duplication, but I think I'd like as much of the info in the docs _also_ available in the issue template/form. Simply because it means the Hub Representative won't have to navigate away to get context/understanding.

I guess I'm also blurring the lines between who is filling in all these details, an engineer or a Hub Representative? If it's an engineer (maybe following a discussion within the opened issue), I guess my above comment could be addressed by a link to the docs you've provided and a prompt to "go here to understand how to help our engineer fill this in" (or something catchier).

File: .github/ISSUE_TEMPLATE/new-hub.md
Comment (choldgraf) [reply .github/ISSUE_TEMPLATE/new-hub.md:34]: Yeah that makes sense assuming that the Community Representative will be the one filling this in. I like framing the question around features ("Do you want XX capability?") rather than implementation ("Do you want to use template XX").

Though for now I think the only feature flag we'd likely add is "Do you want to use a scalable dedicated cluster for Dask?" (since that's the difference between "regular Dask" and "Dask Gateway", I think)

File: docs/howto/operate/new-hub/index.md
Comment (choldgraf) [reply docs/howto/operate/new-hub/index.md:52]: Right so maybe we should add a section to the bottom for the 2i2c Engineer that is deploying the hub to fill in, that would have more 2i2c-specific info in there?

Comment (sgibson91) [reply docs/howto/operate/new-hub/index.md:52]: Yeah, I think so

File: docs/howto/operate/new-hub/index.md
Comment (choldgraf) [reply docs/howto/operate/new-hub/index.md:58]: What do you think about turning this into a GitHub Issue form? That way we could add extra explanations and such from within the form (https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms).

If others think this is a reasonable thing to try, I can try to figure out the new forms syntax and put that together.

File: .github/ISSUE_TEMPLATE/new-hub.md
Comment (choldgraf) [reply .github/ISSUE_TEMPLATE/new-hub.md:33]: yeah good point - I think in this version of the template, there is information that is relevant to the community and to 2i2c engineers intermixed together. What do you think about my comment below regarding using:

- An issue form for the **Community Representative** to fill in
- The opened issue text has a few markdown fields that need to be filled in by a hub engineer (such as cluster name etc)

**Review by choldgraf — COMMENTED** [review]
Oops I think only one of my comments is officially a part of this "review" - but lemme know what you think!

File: docs/howto/operate/new-hub/index.md
Comment (sgibson91) [reply docs/howto/operate/new-hub/index.md:58]: GitHub docs: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

I also transformed some of my issue templates into the forms format too: https://github.com/sgibson91/.github/tree/main/.github/ISSUE_TEMPLATE

**Comment by sgibson91** [comment]
As a general response, I think splitting this into sections that are "to be filled in by the Community Representative" and "to be filled in by a 2i2c engineer" is a good thing. We need all that info in one place, but we need to separate what the Community will care about and what is implementation detail that can be at 2i2c's discretion.

**Comment by choldgraf** [comment]
@sgibson91 whaddaya think about my fancy shmancy new GitHub form?!? :-)

https://github.com/2i2c-org/pilot-hubs/blob/new-hub/.github/ISSUE_TEMPLATE/new-hub.yml

File: .github/ISSUE_TEMPLATE/new-hub.yml
Comment (sgibson91) [inline .github/ISSUE_TEMPLATE/new-hub.yml:33]:
```suggestion
        Should be either a GitHub handle or email address, depending on the type of authentication you've indicated below.
```
The auth field is actually below this one!

File: .github/ISSUE_TEMPLATE/new-hub.yml
Comment (sgibson91) [inline .github/ISSUE_TEMPLATE/new-hub.yml:13]: While I generally approve of line break after every period, unfortunately the issue form doesn't render this as we've come to expect with Markdown. This bugs me on a personal level and shouldn't block this PR, but just raising it in case I'm not alone in my annoyance 😆

File: .github/ISSUE_TEMPLATE/new-hub.yml
Comment (sgibson91) [inline .github/ISSUE_TEMPLATE/new-hub.yml:94]:
```suggestion
      required: false
```
You've stated in the description that this is optional, but the issue form is expecting it to be required 😄

File: .github/ISSUE_TEMPLATE/new-hub.yml
Comment (sgibson91) [inline .github/ISSUE_TEMPLATE/new-hub.yml:113]: This link isn't rendering.

File: .github/ISSUE_TEMPLATE/new-hub.yml
Comment (sgibson91) [inline .github/ISSUE_TEMPLATE/new-hub.yml:139]:
```suggestion
          <!-- The unique ID used to identify this hub. Should be URL-friendly and short. (e.g. `my-university`). -->
```

I would make the descriptions for the fields in the Engineer section HTML comments, so it looks a bit neater when rendered.

**Review by sgibson91 — COMMENTED** [review]
It's really cool! I've left some comments and little bug fixes :)

File: .github/ISSUE_TEMPLATE/new-hub.yml
Comment (GeorgianaElena) [inline .github/ISSUE_TEMPLATE/new-hub.yml:124]: This row is duplicated:
```suggestion
```

**Comment by choldgraf** [comment]
OK I've updated the template per @sgibson91 and @GeorgianaElena's suggestions! I also added a section about preferred cloud provider to make sure that we don't miss that important piece of information if it is there. What do folks think? Is this a good-enough start to try it out with a potential community?

**Comment by sgibson91** [comment]
Screenshot 2021-07-22 at 10 28 55

I think there's some incorrect formatting with the dropdown selection, otherwise I'm +1!

**Review by sgibson91 — APPROVED** [review]

**Comment by choldgraf** [comment]
OK I think I fixed the dropdown (looks like we can't use markdown in those items, which makes sense since it's not really clickable). If you are +1 on it, then I'll merge and send this to a group which wants a hub for a hackathon in a week, and we can see how this goes.

**Comment by choldgraf** [comment]
Will merge this one in so that we can iterate a bit :-)

**Comment by damianavila** [comment]
I like the new form format 😉

**Pull request merged**
